tekton/metadata_test.go

```diff
@@ -0,0 +1,30 @@
+/*
```

Comment:
It _mocks_ the metacopy interface, to avoid influencing tests in the scope of the `tekton` package.

The metadata package functions are NOT tested in `tekton/pipeline_run_test.go`, but rather in the package tests in `metadata/filter_test.go`. 

This kind of thing is a common GO pattern to keep things separated, I guess sorta like OOP in other languages. There are some really cool capabilities this kind of thing allows, for example we could instrument these stub functions to have a `call_counter` , giving the ability for unit tests to do neat things like validating the expected number of function calls. There are packages to do exactly that:
https://pkg.go.dev/github.com/stretchr/testify/mock
https://pkg.go.dev/github.com/golang/mock/gomock
https://github.com/vektra/mockery

Reply:
I don't understand it either tbf. In any case I still think those interfaces and structs should go away, which will remove the need for this file altogether.